File: server/router/shop/shop_notification.go
package shop

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/gin-gonic/gin"
)

type ShopNotificationRouter struct{}

func (s *ShopNotificationRouter) InitShopNotificationRouter(Router *gin.RouterGroup, ClientPrivateRouter *gin.RouterGroup) {
	shopNotificationRouter := Router.Group("shopNotification")
	var api = v1.ApiGroupApp.ShopApiGroup.ShopNotificationApi
	{
		shopNotificationRouter.GET("getAdminNotifications", api.GetAdminNotifications)
		shopNotificationRouter.PUT("markAdminRead", api.MarkAdminRead)
		shopNotificationRouter.PUT("markAllAdminRead", api.MarkAllAdminRead)
		shopNotificationRouter.GET("getAdminUnreadCount", api.GetAdminUnreadCount)
	}

	// 客户端路由（需要 JWT 认证）
	shopNotificationClientRouter := ClientPrivateRouter.Group("shopNotification")
	{
		shopNotificationClientRouter.GET("getMyNotifications", api.GetMyNotifications)
		shopNotificationClientRouter.PUT("markRead", api.MarkRead)
		shopNotificationClientRouter.PUT("markAllRead", api.MarkAllRead)
		shopNotificationClientRouter.GET("getUnreadCount", api.GetUnreadCount)
	}
}

File: web/src/api/shopNotification.js
import service from '@/utils/request'

export const getAdminNotifications = (params) => {
  return service({ url: '/shopNotification/getAdminNotifications', method: 'get', params })
}

export const markAdminRead = (data) => {
  return service({ url: '/shopNotification/markAdminRead', method: 'put', data })
}

export const markAllAdminRead = (data) => {
  return service({ url: '/shopNotification/markAllAdminRead', method: 'put', data })
}

export const getAdminUnreadCount = (params) => {
  return service({ url: '/shopNotification/getAdminUnreadCount', method: 'get', params })
}
